src/net/sourceforge/kolmafia/request/CampgroundRequest.java

```diff
@@ -1265,18 +1285,52 @@ private static boolean parseGarden(final String responseText) {
-            responseText, "rockgarden/c0.gif", ItemPool.WHETSTONE, 0, ItemPool.ROCK_SEEDS, 0)
-        || findImage(
-            responseText, "rockgarden/a1.gif", ItemPool.GROVELING_GRAVEL, 1, ItemPool.ROCK_SEEDS, 1)
-        || findImage(
-            responseText, "rockgarden/b1.gif", ItemPool.MILESTONE, 1, ItemPool.ROCK_SEEDS, 1)
-        || findImage(
-            responseText, "rockgarden/c1.gif", ItemPool.WHETSTONE, 1, ItemPool.ROCK_SEEDS, 1);
+        || findRockGarden(responseText);
+  }
+
+  private static boolean findRockGarden(final String responseText) {
+    if (!responseText.contains("/rockgarden/")) {
+      return false;
+    }
+
+    if (findImage(responseText, "rockgarden/a1.gif", ItemPool.GROVELING_GRAVEL, 1)
+        || findImage(responseText, "rockgarden/a2.gif", ItemPool.GROVELING_GRAVEL, 2)
+        || findImage(responseText, "rockgarden/a3.gif", ItemPool.GROVELING_GRAVEL, 3)
+        || findImage(responseText, "rockgarden/a4.gif", ItemPool.FRUITY_PEBBLE, 1)
+        || findImage(responseText, "rockgarden/a5.gif", ItemPool.FRUITY_PEBBLE, 2)
+        || findImage(responseText, "rockgarden/a6.gif", ItemPool.FRUITY_PEBBLE, 3)
+        || findImage(responseText, "rockgarden/a7.gif", ItemPool.LODESTONE, 1)) {
+      CampgroundRequest.setCampgroundItem(ItemPool.ROCK_SEEDS, 1);
+    } else {
+      findImage(
+          responseText, "rockgarden/a0.gif", ItemPool.GROVELING_GRAVEL, 0, ItemPool.ROCK_SEEDS, 0);
+    }
+
+    if (findImage(responseText, "rockgarden/b1.gif", ItemPool.MILESTONE, 1)
+        || findImage(responseText, "rockgarden/b2.gif", ItemPool.MILESTONE, 2)
+        || findImage(responseText, "rockgarden/b3.gif", ItemPool.MILESTONE, 3)
+        || findImage(responseText, "rockgarden/b4.gif", ItemPool.BOLDER_BOULDER, 1)
+        || findImage(responseText, "rockgarden/b5.gif", ItemPool.BOLDER_BOULDER, 2)
+        || findImage(responseText, "rockgarden/b6.gif", ItemPool.BOLDER_BOULDER, 3)
+        || findImage(responseText, "rockgarden/b7.gif", ItemPool.MOLEHILL_MOUNTAIN, 1)) {
+      CampgroundRequest.setCampgroundItem(ItemPool.ROCK_SEEDS, 1);
+    } else {
+      findImage(responseText, "rockgarden/b0.gif", ItemPool.MILESTONE, 0, ItemPool.ROCK_SEEDS, 0);
+    }
+
+    if (findImage(responseText, "rockgarden/c1.gif", ItemPool.WHETSTONE, 1)
+        || findImage(responseText, "rockgarden/c2.gif", ItemPool.WHETSTONE, 2)
+        || findImage(responseText, "rockgarden/c3.gif", ItemPool.WHETSTONE, 3)
+        || findImage(responseText, "rockgarden/c4.gif", ItemPool.HARD_ROCK, 1)
+        || findImage(responseText, "rockgarden/c5.gif", ItemPool.HARD_ROCK, 2)
+        || findImage(responseText, "rockgarden/c6.gif", ItemPool.HARD_ROCK, 3)
+        || findImage(responseText, "rockgarden/c7.gif", ItemPool.STRANGE_STALAGMITE, 1)) {
+      CampgroundRequest.setCampgroundItem(ItemPool.ROCK_SEEDS, 1);
+    } else {
+      findImage(responseText, "rockgarden/c0.gif", ItemPool.WHETSTONE, 0, ItemPool.ROCK_SEEDS, 0);
```

Comment:
Ditto.

I think you should look for 3 plots images 0-7, add the appropriate plot item, and always add rock seeds (1)

Reply:
The other gardens use the count as the number of days the garden has been planted. We can't do that here because it's actually three gardens.

What I was trying for was to do rock seeds = 0 if the rock garden was present but empty and 1 otherwise. However, I did it wrong :P